driver/src/metrics/http_metrics.rs

```diff
@@ -0,0 +1,249 @@
+use anyhow::Result;
+use prometheus::{IntGauge, IntGaugeVec, Opts, Registry};
+use std::any::{Any, TypeId};
+use std::collections::HashMap;
+use std::fmt::{self, Debug, Formatter};
+use std::sync::Arc;
+use std::time::Duration;
+
+/// A registry for all HTTP related metrics.
+#[derive(Debug)]
+pub struct HttpMetrics {
+    labeled: HashMap<TypeId, SubsystemMetrics<IntGaugeVec>>,
+    unlabeled: HashMap<TypeId, SubsystemMetrics<IntGauge>>,
```

Comment:
I am wondering if the use of `TypeId` is worth it. The simplest solution is to use `&'static str` as the key which we get from the name of the metric. The name of the metric already has to be unique otherwise two metrics would override each other.

Reply:
`TypeId` should be pretty free here. Under the hood its just a `u64` and it should be determined at compile-time and not runtime (since we don't use any `Box<Any>`).